nixos/modules/system/boot/loader/raspberrypi-uboot/raspberrypi.nix

```diff
@@ -0,0 +1,117 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  is64bit = pkgs.stdenv.is64bit;
+
+  cfg = config.boot.loader.raspberryPiUboot;
+
+  uboot = 
+    if cfg.version == 2 then 
+      if is64bit then
+        throw ("Raspberry Pi 2 does not support 64bit")
+      else
+        pkgs.ubootRaspberryPi2
+    else if cfg.version == 3 then
+      if is64bit then
+        pkgs.ubootRaspberryPi3_64bit
+      else
+        pkgs.ubootRaspberryPi3_32bit
+    else
+      throw ("raspberrypi-uboot: unknown version: " ++ cfg.version);
+   
+  configtxt = 
+    pkgs.writeText "config.txt" (if is64bit then
+      ''
+        kernel=u-boot-rpi.bin
+
+        # Boot in 64-bit mode.
+        arm_control=0x200
+
+        # U-Boot used to need this to work, regardless of whether UART is actually used or not.
+        # TODO: check when/if this can be removed.
+        enable_uart=1
+
+        # Prevent the firmware from smashing the framebuffer setup done by the mainline kernel
+        # when attempting to show low-voltage or overtemperature warnings.
+        avoid_warnings=1
```

Comment:
It would be nice to be able to change the content of the `config.txt` via the module option.

Reply:
Yeah, I already thought about that and will integrate that also.